src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/AnalyzerTest`1.cs

```diff
@@ -1263,6 +1269,13 @@ static Solution AddProjectReferences(Solution solution, ProjectId sourceProject,
             {
                 return solution.AddProjectReferences(sourceProject, targetProjects.Select(id => new ProjectReference(id)));
             }
+
+            static (string fileName, IEnumerable<string> folders) GetNameAndFoldersFromPath(string path)
+            {
+                var fileName = Path.GetFileName(path);
+                var folders = Path.GetDirectoryName(path)!.Split("/");
```

Comment:
❓ Is it possible for the path to show up with `\` separators? We use `/` by default in the test framework because it simplifies cross-platform testing, but it seems possible some users are passing in `\` here for individual tests (e.g. by using `Path.Combine`).

Reply:
If users are overriding `DefaultFilePathPrefix` or passing in custom documents, it's possible for `\` to get into the string.